WordPress/gutenberg
# Pattern Directory API: Add support for pagination parameters

This PR enables passing pagination parameters through to the pattern directory API request. In [meta-12147](https://meta.trac.wordpress.org/changeset/12147/), the hard-coded `per_page` and order values were removed, allowing paginated queries. But the connecting API in WordPress/Gutenberg doesn't support those - yet.

The pattern directory explorer here https://github.com/WordPress/gutenberg/pull/45237 will need these parameters to support pagination and ordering patterns using the same sort as wordpress.org/patterns.

This PR sets up an allowed-list of query parameters that are passed through to `api.wordpress.org`. If/when more are needed, they can be added to the list.

## Testing Instructions

1. Ensure the remote patterns are still loading in the editor — the Featured query, the core patterns, and any theme-added patterns.

Test the API endpoint in a REST client.

1. Try pagination, ex: `/wp-json/wp/v2/pattern-directory/patterns/?per_page=6&page=2` — It should show the middle 6 patterns on [wordpress.org/patterns/](https://wordpress.org/patterns/).
2. Try order: `/wp-json/wp/v2/pattern-directory/patterns/?category=37&per_page=3&orderby=date&order=asc` — should show the last 3 items on [wordpress.org/patterns/categories/footer/page/2/](https://wordpress.org/patterns/categories/footer/page/2/)
3. Make sure search still works `/wp-json/wp/v2/pattern-directory/patterns/?search=plants` — should match [wordpress.org/patterns/search/plants/](https://wordpress.org/patterns/search/plants/)
4. Make sure slug still works `/wp-json/wp/v2/pattern-directory/patterns/?slug=centered-text-title-with-two-columns` — should return a single pattern
